### PR TITLE
`GSDumpFile::ReadFile` prealloc

### PR DESCRIPTION
**Pull request by JustinBraben**
### Description of Changes
* Pre-allocate m_packet_data.
* Removes one-off `#define`'s for `GET_BYTE` and `GET_WORD`

### Rationale behind Changes
* Work towards fixing the former `TODO:`.
* Cleans up one-off `#define`'s

### Suggested Testing Steps
* Ensure `GSDumpFile::ReadFile` still works, can add some test code for it.

### CODE REVIEW

File: pcsx2/GS/GSLzma.cpp [hunk 1]
```diff
@@ -126,107 +126,95 @@ bool GSDumpFile::ReadFile(Error* error)
 		return false;
 	}
 
+	// Get remaining file size for packet data preallocation
+    const s64 current_pos = FileSystem::FTell64(m_fp.get());
```
Comment (lightningterror) [inline pcsx2/GS/GSLzma.cpp:130]: Tabs should be used for indentation.

Comment (JustinBraben) [reply pcsx2/GS/GSLzma.cpp:130]: Was working on my macbook and looks like the indents got messed up. Will edit it on my PC !

Comment (JustinBraben) [reply pcsx2/GS/GSLzma.cpp]: Ok I think it is indented properly now @lightningterror

**Comment by JordanTheToaster** [comment]
Crashes when playing back this dump.

[Dump Crash.zip](https://github.com/user-attachments/files/17832522/Dump.Crash.zip)

**Comment by JustinBraben** [comment]
> Crashes when playing back this dump.
> 
> [Dump Crash.zip](https://github.com/user-attachments/files/17832522/Dump.Crash.zip)

Thanks for trying it, will take another look at it in the next few days.

**Comment by JustinBraben** [comment]
This is all pretty new to me, so some direction/explanation would be appreciated.

I built the `pcsx2-gsrunner`, and ran this command against it
```
.\pcsx2-gsrunner\test_run_dumps.py -runner .\bin\pcsx2-gsrunner.exe -dumpdir build\DumpBaseline -gsdir build\DumpCrash -renderer auto
```

It ran the GSDump from the previous comment on my computer and output an `emulog.txt`. It states my computer specs, then says Reading File for the supplied GSDump file. It does successfully change to the correct disc. It runs through emulator commands but I don't see a crash on my end. Have a few questions:

* Is this the correct thing to do with that GSDump file? If not how do I test these out?
* @JordanTheToaster what OS and Rendering backend did you use to get the crash?

My output emulog:  [output_emulog.zip](https://github.com/user-attachments/files/17908452/output_emulog.zip)

**Comment by F0bes** [comment]
This isn't going to work. You're querying the current file pointer and file sizes as if you are reading them as raw binary data, but that isn't the case. You're modifying the Lzma reader which is going to reading in **compressed** data.

The crash JordanTheToaster is getting is due to the fact that `file_size` and `current_pos` are equal because we've already read through the entire Lzma compressed file.
image

**Comment by F0bes** [comment]
Also, you can just drag and drop the GS dumps on to the PCSX2 Qt interface to test if it works or not.

**Comment by JustinBraben** [comment]
Right, I'll take this back to the drawing board then. Thanks for the help!

**Pull request closed without merging**